app/services/policies/policies_list_service.rb

```diff
@@ -4,16 +4,54 @@
 require 'set'
 
 class Policies::PoliciesListService
-  def self.call(account, builtin: true)
-    list = PolicyList.new
-    if builtin
-      response = ::JSONClient.get(ThreeScale.config.sandbox_proxy.apicast_registry_url)
-      return unless response.ok?
 
-      list.merge!(PolicyList.from_hash(response.body['policies']))
-    end
-    list.merge!(PolicyList.new(account.policies)) if account.provider_can_use?(:policy_registry)
+  HTTP_ERRORS = [HTTPClient::BadResponseError, HTTPClient::TimeoutError, HTTPClient::ConnectTimeoutError, HTTPClient::SendTimeoutError, HTTPClient::ReceiveTimeoutError, SocketError].freeze
+  private_constant :HTTP_ERRORS
+
+  class PoliciesListServiceError < RuntimeError; end
+
+  def self.apicast_registry_url
+    ThreeScale.config.sandbox_proxy.apicast_registry_url
+  end
+
+  delegate :apicast_registry_url, to: 'self.class'
+
+  def self.call(*args)
+    new(*args).call
+  end
+
+  def initialize(account, builtin: true)
+    @account = account
+    @builtin = builtin
+  end
+
+  attr_reader :account, :builtin
+  alias builtin? builtin
+
+  def call
+    list = PolicyList.new
+    list.merge! fetch_policies_from_apicast if builtin?
+    list.merge! policies_from_account
     list.to_h
+  rescue *HTTP_ERRORS, PoliciesListServiceError => error
+    Rails.logger.error { error } and return
```

Comment:
wouldn't be better to return blank hash `and return {}` ?

i mean, then you wouldn't have to check if it's `nil` but just do the `each` 

cc @Martouta

Reply:
I don't know why, this is a question for @josemigallas , but @hallelujah said something similar in https://github.com/3scale/porta/pull/736#discussion_r275671274 and @josemigallas said that it needs to return `nil` and that there was already a test for it

Reply:
I also find it strange (not necessarily incorrect), but it is true that we already have a test in the master branch that checks the response is `nil` in case of error.
https://github.com/3scale/porta/blob/50988a46fc210ae75f1f3edea3fff013f2c3c612/test/unit/services/policies/policies_list_service_test.rb#L25-L32

Reply:
Yes the reason is simply the definition was to return nil so I kept it like that

Reply:
ok, make sense 🍳